### DISCUSSION TITLE
四元數

### DISCUSSION THREAD

**Discussion by 4rfvedc** [1101-GS4719-C/Python-Community-GS4719-C#1481] [閒聊八卦]
首先，讓我們來了解一下四元數的基本概念啊。四元數是由實數和虛數單位 i、j、k 構成的。和複數不同的是，四元數有兩個虛數單位 j 和 k，它們滿足一些特殊的乘法規則。

進一步來說，四元數可以表示為 a + bi + cj + dk 的形式，其中 a、b、c、d 是實數部分。這樣的表示法可以幫助我們進行四元數的運算。

四元數的運算包括加法和乘法兩種。加法是直接將對應位置的實數和虛數部分相加。而乘法則涉及到虛數單位 i、j、k 的特殊規則。這些運算讓我們能夠在四元數空間中進行更多的數學操作呀。

四元數的應用非常廣泛！在物理學、電腦圖形學和機器人學等領域中，四元數被廣泛地應用於旋轉、定向和運動控制等問題上。其獨特的性質使得它成為一個非常重要且強大的工具。

雖然四元數可能對許多人來說有點抽象和複雜，但它卻在數學和應用領域中扮演著重要的角色。通過學習和理解四元數，我們可以更深入地探索數學的奧秘，並在各個領域中運用它們來解決問題。

**Comment by KozakHou** [comment]
在太空系裡面我是做衛星姿態控制的。
其實四元數是所有衛星、火箭任何到達太空中的物件都必須要有的。尤其衛星自主要去藉由太陽感測器，搭配衛星航電去讓控制器去調適衛星太陽能板正對太陽的位置，抑或是使用星象儀(裝在衛星上的攝影機，用來拍星星)，來辨識自身位置，這個時候如何去知道所在位置跟如何調整成任務模式還是充電模式都會使用到四元數進行調整。
感謝你的分享，總算有討論是可以跟太空相關領域做連結。

**Comment by pudding0803** [comment]
之前沒有聽過，感覺好酷😮，不過就是不知道能應用在哪裡，數學總是有一些很神奇的用途🤔

在這邊補充一下，這是我好奇上網查到的乘法對應表：
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/50cfad44-b7f8-487a-b0a5-581952f1ffc9)

他們的乘法是沒有交換律的！

**Comment by RogelioKG** [comment]
上面那張乘法表很酷，
我記得之前有看過 QRcode 機制的詳細介紹。
裡面也引用了「伽羅瓦域」的運算特性 (裡面的數運算是封閉的)。
使得我們能運用糾錯碼求出錯誤位置與與錯誤誤差。
伽羅瓦域中加法和減法都是異或運算，
而乘法 (GF(2^3)) 則是這張表：

![353156799_668766168424045_4473796176028475022_n](https://github.com/NCU-GS4719-Python/Python-Community/assets/86526364/3d4eabdd-4d58-4f06-bf35-569482b08709)
